hail/hail/src/is/hail/expr/ir/TableIR.scala

```diff
@@ -1509,6 +1509,40 @@ case class PartitionZippedNativeReader(left: PartitionReader, right: PartitionRe
   }
 }
 
+case class PartitionZippedNativeIntervalReader(
+  sm: HailStateManager,
+  mtPath: String,
+  mtSpec: AbstractMatrixTableSpec,
+  uidFieldName: String,
+) extends PartitionReader {
+  require(mtSpec.indexed)
+
+  // XXX: rows and entries paths are hardcoded, see MatrixTableSpec
+  private val rowsReader = PartitionNativeIntervalReader(sm, mtPath + "/rows", mtSpec.rowsSpec, "__dummy")
+  private val entriesReader = PartitionNativeIntervalReader(sm, mtPath + "/entries", mtSpec.entriesSpec, uidFieldName)
+  private val zippedReader = PartitionZippedNativeReader(rowsReader, entriesReader)
+
+  def contextType = rowsReader.contextType
+  def fullRowType = zippedReader.fullRowType
```

Comment:
@patrick-schultz I'm getting an NPE in initialization, I think it's because this assertion in PartitionReader is running before `zippedReader` is initialized. Thoughts on resolving that?
https://github.com/hail-is/hail/blob/ecaa96bc4c869fe3b2b4474e7f04a12958e78c0b/hail/hail/src/is/hail/expr/ir/IR.scala#L1300-L1301

Reply:
That's annoying. My first thought is to make the sub-readers lazy vals. Maybe try that for now, and we can think more if there's any more satisfying fix.

Reply:
Thanks for the suggestion. It worked. I guess `lazy val` can force initialization, but `val` is only truly valid after full initialization of the class.